include/tfrt/support/aligned_buffer.h

```diff
@@ -36,11 +36,18 @@ struct AlignedAllocator {
 
   using value_type = T;
 
+  constexpr AlignedAllocator() noexcept = default;
+
+  constexpr AlignedAllocator(const AlignedAllocator&) noexcept = default;
+
+  template <class Other>
+  constexpr AlignedAllocator(const AlignedAllocator<Other>&) noexcept {}
+
   T* allocate(size_t n) {
     auto* ptr = AlignedAlloc(Align, n * sizeof(T));
     return static_cast<T*>(ptr);
   }
-  void deallocate(T* p, size_t) noexcept { std::free(p); }
+  void deallocate(T* p, size_t) noexcept { AlignedFree(p); }
```

Comment:
Good questions!

The `noexcept` was there in the original code and I left it alone, although it's not really required for conformance to the allocator interface. Since you mentioned, it's better to remove it (to avoid suboptimal code generation in this case)

Regarding the need for `AlignedFree()` vs. just calling `std::free()`: On may unix implementation, the aligned allocations can be freed with `free()`. This is not always the case, and in particular with the MSVC runtime, `_aligned_malloc` must be paired with `_aligned_free()` (trying to free aligned blocks with `free()` results in nasty runtime issues)

Finally, the additional constructors are a required part of the allocator interface. MSVC debug runtime needs these constructors for implementing some of the debug checks.

Reply:
Thanks for the explanation. Would you mind leaving some breadcrumb comments in the code about this?